hazelcast/src/main/java/com/hazelcast/cardinality/impl/CardinalityEstimatorContainer.java

```diff
@@ -26,12 +26,19 @@
 
 public class CardinalityEstimatorContainer implements IdentifiedDataSerializable {
 
+    private int durability;
+
     private HyperLogLog hll;
 
     public CardinalityEstimatorContainer() {
         hll = new HyperLogLogImpl();
```

Comment:
Why don't you create the HyperLogImpl when the variable is initialized? I normally try to do this to keep the constructors as small as possible.

Once this is done, you can create a single master constructor and let the other constructors just be convenience implementations.

```
public CardinalityEstimatorContainer(){
  this(0);
}

public CardinalityEstimatorContainer(int durability){
    ....
}
```